[RESTEASY-3018] Vertx add query params to request and improve implementation for Quarkus.

OscarRL opened the pull request:
This implementation makes it easy use the RestEasy Client with Quarkus Vertx.

- Adds a fix to add query params on request.
- Changed uri handling to pass the absolute uri to HttpClient because normal URI fails to resolve port if not specified.
- Check if user agent exists to avoid duplicate agent. 
- Force add vertx handler to use in case that there is no executor service to deserialize request.

OscarRL commented:
+ Changed uri handling to pass the absolute uri to HttpClient because normal URI fails to resolve port if not specified.

jamezp reviewed (commented):
We require all contributions to be made under the terms of the ASL 2.0 License: http://www.apache.org/licenses/LICENSE-2.0 
Do you agree to these terms?

OscarRL commented:
Yes, I expect that. But I did not found any form to apply to the licence on your website.

OscarRL commented:
Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX 

https://vertx.io/docs/vertx-web-client/java/

This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.

We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?

OscarRL commented on resteasy-client-vertx/pom.xml:
```diff
@@ -60,6 +60,11 @@
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-rxjava2</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Solved

jamezp commented:
> Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX
> 
> https://vertx.io/docs/vertx-web-client/java/
> 
> This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.
> 
> We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?

To be completely honest we (I?) have been in purge mode with RESTEasy recently. I think we've got too many projects that are mostly independent of RESTEasy.

Some of these things we've moved to new projects. Some of them we moved to https://github.com/resteasy/resteasy-extensions.

That said I'm definitely not opposed to new things like this. I just think finding the right home is key and I'm absolutely open for discussions about it. We could start a discussion https://github.com/resteasy/resteasy/discussions if you'd like.

OscarRL commented:
> > Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX
> > https://vertx.io/docs/vertx-web-client/java/
> > This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.
> > We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?
> 
> To be completely honest we (I?) have been in purge mode with RESTEasy recently. I think we've got too many projects that are mostly independent of RESTEasy.
> 
> Some of these things we've moved to new projects. Some of them we moved to https://github.com/resteasy/resteasy-extensions.
> 
> That said I'm definitely not opposed to new things like this. I just think finding the right home is key and I'm absolutely open for discussions about it. We could start a discussion https://github.com/resteasy/resteasy/discussions if you'd like.

yes but I think that Red Hat is focusing the development to Quarkus. And a good VertX core for the http should integrate better than the Apache HttpCore 4, that currently is outdated cause Apache released the version 5.

jamezp commented:
> yes but I think that Red Hat is focusing the development to Quarkus. And a good VertX core for the http should integrate better than the Apache HttpCore 4, that currently is outdated cause Apache released the version 5.

I can definitely say that WildFly is a heavy focus as well :) I have however suggested that we start looking at deprecating the Apache HTTP Client integration in favor of Vert.x. This will like take some time to happen though as we also need to focus on Jakarta REST 3.0 and 3.1.

OscarRL commented:
Updated master branch and tests solved.

jamezp commented:
Thank you for the contribution. We are cleaning up older PR's that seem to be stale. If you feel this is still an issue please feel free to re-open.
